GPy/testing/inference_tests.py

```diff
@@ -132,6 +133,16 @@ def test_inference_EP_non_classification(self):
                     np.sum(p._woodbury_vector - p0._woodbury_vector),
                     np.sum(p.woodbury_inv - p0.woodbury_inv)])) < 1e6)
 
+class VarDtcTest(unittest.TestCase):
+
+    def test_var_dtc_inference_with_mean(self):
+        """ Check dL_dm in var_dtc is calculated correctly"""
+        np.random.seed(1)
+        x = np.linspace(0.,2*np.pi,100)[:,None]
+        y = -np.cos(x)+np.random.randn(*x.shape)*0.3+1
+        m = GPy.models.SparseGPRegression(x,y, mean_function=GPy.mappings.Linear(input_dim=1, output_dim=1))
+        self.assertTrue(m.checkgrad())
+
```

Comment:
Can we have a unittest that verify the prediction of sparse GP with mean function?

Reply:
Hmm, good point I hadn't thought of that. Unless I'm missing them we don't have such tests for the vanillia GP prediction function either (which we should)? Or do you just mean testing that the prediction function doesn't throw an exception? As in:
https://github.com/SheffieldML/GPy/blob/83c3ed44e28b5996e262fcaba18d612d63d6e274/GPy/testing/gp_tests.py#L96